pkg/config/model/viper.go

```diff
@@ -235,6 +250,7 @@ func (c *safeConfig) AllKeysLowercased() []string {
 func (c *safeConfig) Get(key string) interface{} {
 	c.RLock()
 	defer c.RUnlock()
+	c.checkKnownKey(key)
```

Comment:
Could we benchmark this PR before merging. Since `Get` is called in some hot path in the Agent it would be nice to get a idea of the performance impact.

Reply:
I'm not super sure what to benchmark, just the duration of a call to `Get`, or the overall impact on the agent ?
I'll make a quick benchmark specifically around the `Get` function and update the description with the results

The implementation change of `IsKnown` was specifically to reduce the overhead of that check, I agree having a benchmark to evaluate the final impact would be nice 👌

Reply:
I updated the description with some measurements:
- for the happy path it adds 5ns/op, I've observed before that a map access is usually around 2ns so that's what I expected
- for the "bad" path it increases the duration by an order of magnitude, since it needs to `RUnlock`, `Lock`, `Unlock`, `log`, and `RLock`
The "bad" path can happen only once per config key, and overall is not supposed to happen at all (it highlights a key that should be declared but isn't), so I don't think this is an issue

Reply:
Thanks for doing this :+1: